unitstatushandler/softwaremanager.go

```diff
@@ -823,11 +828,72 @@ func (manager *softwareManager) update(ctx context.Context) {
 		updateErr = err
 	}
 
-	if err := manager.runInstances(newServices); err != nil && updateErr == nil {
+	if err := manager.runInstances(); err != nil && updateErr == nil {
 		updateErr = err
 	}
 
-	manager.runCond.Wait()
+	manager.runStatusCV.Wait()
+
+	if len(manager.revertServices) != 0 {
+		manager.doRevertServices()
+		manager.revertServices = nil
+
+		if err := manager.runInstances(); err != nil && updateErr == nil {
```

Comment:
Before starting service it should be installed, I guess. But just reverting the service should be enough as the first step for recovering instance run.

Rerunning the instances can require more difficult strategy to implement.

Reply:
For example?